weasyprint/tests/test_layout/test_table.py

```diff
@@ -1536,6 +1536,36 @@ def test_layout_table_auto_50():
     assert td_21.width == 240  # 15 * font_size
 
 
+@assert_no_logs
+@pytest.mark.parametrize('size, width', (
+    ('242px', 242),
+    ('100%', 793),
+))
+def test_explicit_width_table_percent_rtl(size, width):
+    page, = render_pages('''
+      <table style="width: %s; direction:rtl">
+        <tr>
+          <th>الاسم</th>
+          <th>العائلة</th>
+          <th>العمر</th>
+        </tr>
+        <tr>
+          <td>محمد يوسف</td>
+          <td>النجدي</td>
+          <td>29</td>
+        </tr>
+      </table>
+    ''' % size)
+    html, = page.children
+    body, = html.children
+    wrapper, = body.children
+    table, = wrapper.children
+    row_group, = table.children
+
+    assert table.position_x == 0
+    assert int(table.width) == width
+
```

Comment:
We have to test the positions of the columns as well.

Reply:
I will test them and let you know.